examples/README.md

```diff
@@ -0,0 +1,159 @@
+
+```
+#!/bin/bash -e
+#
+# S2I assemble script for the 'nginx-centos7' image.
+# The 'assemble' script builds your application source ready to run.
+#
+# For more information refer to the documentation:
+#	https://github.com/openshift/source-to-image/blob/master/docs/builder_image.md
+#
+
+if [[ "$1" = "-h" ]]; then
+	# If the 'nginx-centos7' assemble script is executed with '-h' flag,
+	# print the usage.
+	exec /usr/libexec/s2i/usage
+fi
+
+# Restore artifacts from the previous build (if they exist).
+# We set them here just for show, but you will need to set this up with your logic
+# according to the application directory that you chose.
+if [ "$(ls /tmp/artifacts/ 2>/dev/null)" ]; then
+  echo "---> Restoring build artifacts..."
+  mv /tmp/artifacts/* /usr/share/nginx/html/
+fi
+
+# Override our default nginx index.html file.
+# This is what we consider in this example `installing the application`
+# here you can go ahead an replace this with the actual building of python modules,
+# bundle install, and anything else you need.
+echo "---> Installing application source..."
+cp -Rf /tmp/src/test/index.html /usr/share/nginx/html/index.html
+```
+What we need to do is tell the run script what process to run on start.  
+
+```
+#!/bin/bash -e
+# The run script executes the server that runs your application.
+#
+# For more information see the documentation:
+#	https://github.com/openshift/source-to-image/blob/master/docs/builder_image.md
+#
+
+# We will turn off daemonizing for the nginx process so that the container
+# doesn't exit after the process runs.
+# We will also specify the default port on which it should listen.
+exec /usr/sbin/nginx -g "daemon off;"
+```
+
+We can also specify a logic for an incremental build in `save-artifacts`.    
+
+```
+#!/bin/sh -e
+#
+# S2I save-artifacts script for the 'nginx-centos7' image.
+# The save-artifacts script streams a tar archive to standard output.
+# The archive contains the files and folders you want to re-use in the next build.
+#
+# For more information see the documentation:
+#	https://github.com/openshift/source-to-image/blob/master/docs/builder_image.md
+#
+# Replace this with any logic that you need in order to save all of your built artifacts from your application.
+touch /tmp/artifact
+cd /tmp
+tar cf - artifact
+```
+
+We are now ready to create our base image that contains nginx and we will name it `nginx-centos7`.
+
+`docker build -t nginx-centos7 .`
+
+With the base image created we can now create the builder image.
+More [here](https://github.com/openshift/source-to-image/blob/master/docs/builder_image.md).
+
+```
+s2i build test/ nginx-centos7 s2i-nginx
+---> Installing application source
+```
+Printing out the usage format of the image
+
+```
+s2i usage s2i-nginx
```

Comment:
this should be s2i usage nginx-centos7, because nginx-centos7 is the s2i builder image.

s2i-nginx is a (badly named) application image.  should call it nginx-app or something instead.